### PR TITLE
refactor: allow specifying multiple channel ids instead of one

### PR DESCRIPTION
**Pull request by jackstar12**
closes: #43

### CODE REVIEW

File: cln/cln.go [hunk 1]
```diff
@@ -192,18 +193,18 @@ func (c *Cln) CreateInvoice(value int64, preimage []byte, expiry int64, memo str
...
 }
 
-func (c *Cln) PayInvoice(invoice string, feeLimit uint, timeoutSeconds uint, chanId lightning.ChanId) (*lightning.PayInvoiceResponse, error) {
+func (c *Cln) PayInvoice(invoice string, feeLimit uint, timeoutSeconds uint, chanIds []lightning.ChanId) (*lightning.PayInvoiceResponse, error) {
 	retry := uint32(timeoutSeconds)
 
 	var exclude []string
 
-	if chanId != 0 {
+	if len(chanIds) > 0 && chanIds[0] != 0 {
```
Comment (michael1011) [inline cln/cln.go:201]:
```suggestion
	if len(chanIds) > 0 {
```
Why do we still need this check?

File: lnd/lnd.go [hunk 2]
```diff
@@ -282,28 +282,23 @@ func (lnd *LND) GetChannelInfo(chanId uint64) (*lnrpc.ChannelEdge, error) {
...
+func (lnd *LND) PayInvoice(invoice string, feeLimit uint, timeoutSeconds uint, chanIds []lightning.ChanId) (*lightning.PayInvoiceResponse, error) {
+	var outgoungIds []uint64
+	for _, chanId := range chanIds {
+		if chanId != 0 {
+			outgoungIds = append(outgoungIds, uint64(chanId))
 		}
-	*/
+	}
+
+	fmt.Println(outgoungIds)
```
Comment (michael1011) [inline lnd/lnd.go:293]: Rm

File: lnd/lnd.go [hunk 3]
```diff
@@ -282,28 +282,23 @@ func (lnd *LND) GetChannelInfo(chanId uint64) (*lnrpc.ChannelEdge, error) {
...
 
 	client, err := lnd.router.SendPaymentV2(lnd.ctx, &routerrpc.SendPaymentRequest{
 		PaymentRequest:    invoice,
 		TimeoutSeconds:    int32(timeoutSeconds),
 		FeeLimitSat:       int64(feeLimit),
 		NoInflightUpdates: true,
 		// TODO: this is not working right now
```
Comment (michael1011) [inline lnd/lnd.go:300]: This you manage to reliable reproduce the flakiness?

**Review by michael1011 — CHANGES_REQUESTED** [review]

File: cln/cln.go (hunk 1 above)
Comment (jackstar12) [reply cln/cln.go:201]: Autoswap recommendations include a channel id, which is passed to the create swap call, but the zero value means no channel. I guess that check could be put somewhere else

File: lnd/lnd.go (hunk 3 above)
Comment (jackstar12) [reply lnd/lnd.go:300]: In regtest:
```bash
clnId=$(lightning-cli-sim 1 getinfo | jq -r  .id)
chanId=$(lncli-sim 1 listchannels --peer $clnId | jq -r  ".channels.[].chan_id")
./boltzcli createreverseswap --chan-id $chanId  100000
```

Comment (jackstar12) [reply lnd/lnd.go]: its a regtest issue, works in testnet

**Review by michael1011 — APPROVED** [review]
utACK

**Pull request merged**